gprofiler/py_package_version.py

```diff
@@ -0,0 +1,160 @@
+"""Package name and version for Python files.
+
+Some of the functions in this module are implemented based on similar functions
+in pip: _get_metadata, _convert_legacy_entry, _files_from_record,
+_files_from_legacy, _get_package_name.
+"""
+import csv
+import email
+import os
+import pathlib
+from typing import Iterator, List, Optional, Tuple
+
+import pkg_resources
+
+# A little monkey patch to prevent pkg_resources from converting "/proc/{pid}/root/" to "/"
+pkg_resources._normalize_cached = lambda path: path
+
+
+__all__ = ["get_versions"]
+
+
+def _convert_to_proc_root_path(path: str, pid: int) -> str:
+    assert path.startswith("/")
+    return os.path.join(f"/proc/{pid}/root", path)
+
+
+def _get_packages_dir(file_path: str) -> Optional[str]:
+    if not file_path.startswith("/"):
+        return None
+
+    idx = file_path.rfind("-packages/")
+    if idx == -1 or (not file_path[:idx].endswith("site") and not file_path[:idx].endswit("dist")):
+        return None
+
+    return file_path[:idx] + "-packages/"
+
+
+def _get_metadata(dist: pkg_resources.Distribution) -> email.message.Message:
```

Comment:
Just a dict-like object that metadata files can be parsed into. Actually I could equivalently cast the result to `dict` before returning it, it would've been clearer.

Reply:
Good idea with the dict, this `email.message.Message` isn't known to anyone

Reply:
Is this function is taken from pip/setuptools? If so please specify

Reply:
Based on a function from there, I'll add documentation for all of them.